### PR TITLE
chore: Introduces TransactionOptionsInterface

### PR DESCRIPTION
**Pull request by JirkaChadima**
Improves type-checking and fixes some inconsistencies.

### CODE REVIEW

File: src/data-model/on-chain-hotel.js [hunk 1]
```diff
@@ -252,11 +252,11 @@ class OnChainHotel implements HotelInterface {
    * Winding Tree index contract.
    *
    * @param {WalletInterface} wallet that signs the transaction
-   * @param {Object} transactionOptions usually contains from and to. Gas is automatically computed.
+   * @param {TransactionOptionsInterface} transactionOptions usually contains from and to. Gas is automatically computed.
```
Comment (MatiasOS) [inline src/data-model/on-chain-hotel.js]: What do you think about something like "must have `from`, and optionally `to`, `gas` and `value`"

Comment (JirkaChadima) [reply src/data-model/on-chain-hotel.js]: They are optional. That's defined i the interface with gas?  Yhe question mark means optional

**Comment by JirkaChadima** [comment]
@MatiasOS I've changed the docstrings and realized that we're actually not using `to` property at all, so I removed that as well. Is the meaning more clear now?

**Comment by JirkaChadima** [comment]
Ping @MatiasOS

**Comment by MatiasOS** [comment]
Yes, I like it now. I got confused by the difference between the definition of the interface and the comment. It seems much clearer to me!

**Pull request merged**